opentelemetry-sdk/src/metrics/pipeline.rs

```diff
@@ -312,6 +313,10 @@ where
             });
         }
 
+        if let Some(cardinality_limit) = cardinality_limit {
```

Comment:
nit: We could simply assign the cardinality limit when creating the stream in line 299.

```rust
// Apply implicit default view if no explicit matched.
        let mut stream = Stream {
            name: inst.name,
            description: inst.description,
            unit: inst.unit,
            aggregation: None,
            allowed_attribute_keys: None,
            cardinality_limit: cardinality_limit,
        };
```